### PR TITLE
[#3645] Create new endpoints for the app to retrieve datapoints in smaller chunks

### PR DESCRIPTION
**Pull request by muloem**
#### Before the PR (what is the issue or what needed to be done)
* The backend would Time out when there are lots of form instances associated with a specific data point

#### The solution
* Split the retrieval of datapoints into smaller chunks. One endpoint returns only the datapoints and a second one returns all form instances (paginated) for a single datapoint.

#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/app/web/SurveyInstanceServlet.java [hunk 1]
```diff
@@ -85,14 +90,21 @@ protected RestResponse handleRequest(RestRequest req) throws Exception {
...
+            try {
+                List<SurveyInstance> formInstances = formInstanceUtil.getFormInstances(siReq.getAndroidId(), siReq.getDataPointIdentifier(), LIMIT_FORM_INSTANCES_30, siReq.getCursor());
+                response.setSurveyInstances(formInstanceUtil.getFormInstancesDtoList(formInstances));
+                response.setCursor(BaseDAO.getCursor(formInstances));
+                response.setResultCount(formInstances.size());
+
+                return response;
+            } catch (Exception e) {
+                log.warning("Exception accessing endpoint: " + e.getMessage());
+                response.setCode(String.valueOf(HttpServletResponse.SC_NOT_FOUND));
```
Comment (tangrammer) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyInstanceServlet.java:105]: not sure if every exception is a not found exception 🤔 ... maybe we need a couple of specific exceptions to know which kind was thrown (eg: DeviceNotFoundException) from the formInstanceUtil

File: GAE/src/org/waterforpeople/mapping/app/web/SurveyInstanceServlet.java [hunk 2]
```diff
@@ -18,14 +18,17 @@
...
 
 import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
 import com.gallatinsystems.framework.dao.BaseDAO;
 import org.akvo.flow.api.app.FormInstanceResponse;
 import org.akvo.flow.api.app.FormInstanceUtil;
 import org.akvo.flow.util.FlowJsonObjectWriter;
 import org.springframework.beans.BeanUtils;
+import org.springframework.http.HttpStatus;
```
Comment (iperdomo) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyInstanceServlet.java:31]: We're not using this class. Let's remove the import.

Comment (iperdomo) [reply GAE/src/org/waterforpeople/mapping/app/web/SurveyInstanceServlet.java]: Change made and pushed

File: GAE/src/org/waterforpeople/mapping/app/web/SurveyInstanceServlet.java (hunk 1 above)
Comment (iperdomo) [reply GAE/src/org/waterforpeople/mapping/app/web/SurveyInstanceServlet.java:104]: then you enter in the *longish conversation* of checked vs unchecked exceptions. https://www.baeldung.com/java-checked-unchecked-exceptions

**Review by iperdomo — APPROVED** [review]

**Pull request merged**